Proxy/HTTP/NotesKeywordHighlighter.bambda

```diff
@@ -0,0 +1,29 @@
+/**
+ * Notes Filter Bambda
+ * This Bambda filters Proxy HTTP history for entries with notes containing a specified keyword.
+ * It checks for a specific keyword within the notes and highlights the matching entries.
+ * Users can easily modify the 'keyword' variable to suit their specific search criteria.
+ * Author: Tur24Tur
+ * GitHub: @BugBountyzip (https://github.com/BugBountyzip)
+ **/
+
+// User-defined keyword for filtering notes
+String keyword = "High"; // Replace "High" with your desired keyword
+
+// Main logic of the Bambda
+if (requestResponse.request().url() != null && requestResponse.hasResponse()) {
```

Comment:
Do we need the if statement given that we're only checking for annotations?

Reply:
yes because if the requests has no response the script will not work on your burp history there are some requests with no response

Reply:
Please feel free to modify this or suggest any improvements. If you require assistance, do not hesitate to reach out




I also noticed that searching for 'High' does not yield results for 'high' due to case sensitivity. Would you prefer this to be fixed, or should we consider it a feature?

![image](https://github.com/PortSwigger/bambdas/assets/133497067/216ba7a3-791b-4bfc-88a4-13c827e65405)

Reply:
For this particular bambda, you're not looking at any response data so the conditional seems a little unnecessary. I think the exceptions you have seen elsewhere are due to attempting to access responses from proxy history items for which the response has no been received yet, i.e. it's fine to access **_requestResponse.annotations()_** when there is no response.

With respect to the case sensitivity, you're working with Java String objects which are case sensitive. If you want case insensitive matching then one way to achieve this would be to lower case both the notes and the keyword before the check.

Reply:
let me check


Thank you

Reply:
Dear [PortSwiggerWiener](https://github.com/PortSwiggerWiener)

I have made several enhancements to the Notes Filter Bambda script. The key updates are as follows:

 Case Sensitivity Toggle - >  Implemented a boolean flag caseSensitive to allow users to choose between case-sensitive and case-insensitive searches for keywords in notes. This should address the issue where different case variations of keywords were not being identified effectively.

and more enhancement  <3